src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,504 @@
+#
+#     f = mill_features(millings, sites, feature_post_status, acq_streams, ebeam, sed, stage)
+#     # gui - add_done_Callback()
+#     f.result()
+
+
+class MillingSettings(object):
+    """
+    Model class for milling settings
+
+    """
+
+    def __init__(self, name, current, horizontal_fov, roi,
+                 pixel_size, beam_angle, duration,
+                 dc_roi, dc_period, dc_dwell_time, dc_current):
+        """
+       Settings class for milling a rectangle.
+       
+       :param name: (str) name for the given milling setting.
+       :param current: allow currents from the predefined
+       :param horizontal_fov:  do not ask user, keep it fixed below 45um
+       :param roi: 
+       :param pixel_size:  do not ask user, it needs to set by us
+       :param beam_angle: 
+       :param duration: 
+       :param dc_roi: 
+       :param dc_period: 
+       :param dc_dwell_time:  do not ask user, depends on dc_current
+       :param dc_current:  do not ask user, keep it same as milling current
+       :return: 
+       """
+        current_choices = {20e-12, 100e-12, 350e-12, 760e-12}
+        dwell_time_choices = {1e-6, 10e-6} # for drift correction
+        self.name = model.StringVA(name)
+        self.current = model.FloatEnumerated(current, unit="A",
+                                             choices=current_choices)
+        self.horizontalFoV = model.FloatContinuous(horizontal_fov, unit="m", range=(5e-06, 45e-06))
+        self.roi = model.TupleContinuous(roi,
+                                         range=((0, 0, 0, 0), (1, 1, 1, 1)),
+                                         cls=(int, long, float))
+        self.pixelSize = model.VigilantAttribute(pixel_size, unit="m")
+        self.beamAngle = model.FloatContinuous(beam_angle, unit="rad", range=(math.radians(6), math.radians(10)))  # in degrees
+        self.duration = model.FloatVA(duration, unit="s")
+        # drift correction settings
+        self.dcRoi = model.TupleContinuous(dc_roi,
+                                           range=((0, 0, 0, 0), (1, 1, 1, 1)),
+                                           cls=(int, long, float))
+        self.dcPeriod = model.FloatContinuous(dc_period, unit="s", range=(1, 180))
+        self.dcDwellTime = model.FloatEnumerated(dc_dwell_time, unit="s", choices=dwell_time_choices)
+        self.dcCurrent = model.FloatEnumerated(dc_current, unit="A",
+                                               choices=current_choices)
+
+
+# def load_config(yaml_filename):
+#     """Load user input from yaml settings file.
+#     Parameters
+#     ----------
+#     yaml_filename : str
+#         Filename path of user configuration file.
+#     Returns
+#     -------
+#     dict
+#         Dictionary containing user input settings.
+#     """
+#     with open(yaml_filename, "r") as f:
+#         settings_dict = yaml.safe_load(f)
+#
+#     # settings = Settings(**settings_dict)  # convert to python dataclass
+#     return settings_dict
```

Comment:
```suggestion
# def load_config(yaml_filename):
#     """Load user input from yaml settings file.
#     Parameters
#     ----------
#     yaml_filename : str
#         Filename path of user configuration file.
#     Returns
#     -------
#     dict
#         Dictionary containing user input settings.
#     """
#     with open(yaml_filename, "r") as f:
#         settings_dict = yaml.safe_load(f)
#
#     # settings = Settings(**settings_dict)  # convert to python dataclass
#     return settings_dict
```

def load_config(yaml_filename):
    """Load user input from yaml settings file.
    Parameters
    ----------
    yaml_filename : str
        Filename path of user configuration file.
    Returns
    -------
    dict
        Dictionary containing user input settings.
    """
    try:
        with open(yaml_filename, "r") as f:
            settings_dict = yaml.safe_load(f)

    except yaml.YAMLError as exc:
        logging.error("Syntax error in milling settings yaml file: %s", exc)

    # settings = Settings(**settings_dict)  # convert to python dataclass
    return settings_dict